chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -229,11 +229,14 @@ async def puzzle_solution_received(self, coin_spend: CoinSpend, in_transaction:
             if condition_code == -24:
                 # metadata update
                 # (-24 (meta updater puzzle) url)
-                metadata_dict = dict(metadata.as_python())
-                new_metadata = [
-                    (b"u", ([condition.rest().rest().first().atom] + [metadata_dict[b"u"]])),
-                    (b"h", metadata_dict[b"h"]),
-                ]
+                metadata_list = list(metadata.as_python())
+                new_metadata = []
+                for metadata_entry in metadata_list:
+                    key = metadata_entry[0]
+                    if key == b"u":
+                        new_metadata.append((b"u", [condition.rest().rest().first().atom] + list(metadata_entry[1:])))
```

Comment:
Why list(metadata_entry[1:]) here? Shouldn't be metadata_entry[1]? Since it's a list of uris